core/src/optimistic_confirmation_verifier.rs

```diff
@@ -85,6 +85,9 @@ impl OptimisticConfirmationVerifier {
                     );
                 }
                 datapoint_info!("optimistic_slot", ("slot", new_optimistic_slot, i64),);
+                if let Some(metrics) = agave_metrics::singleton_metrics::get_metrics() {
+                    metrics.core.optimistic_slot.set(new_optimistic_slot as i64);
+                }
```

Comment:
So as written, the callsite for each datapoint would need to be updated. Not necessarily a deal-breaker, but as seamless as updating something inside the macro definition

Reply:
I’m not sure if it’s possible to create a convenient macro. we will need to register these datapoints first. I’ll do more investigation to see if we can register the datapoints later. if we can, then maybe a macro is possible. btw, the current macro mechanism spreads the datapoints everywhere. It’s a bit difficult to know if we have some datapoints for something in one place